source/world/level/Level.cpp

```diff
@@ -304,37 +283,37 @@ Entity* Level::getEntity(int id)
 	return nullptr;
 }
 
-EntityVector* Level::getAllEntities()
+const EntityVector* Level::getAllEntities() const
 {
 	return &m_entities;
 }
 
-bool Level::hasChunk(int x, int z)
+bool Level::hasChunk(int x, int z) const
 {
 	return m_pChunkSource->hasChunk(x, z);
 }
 
-EntityVector* Level::getEntities(Entity* pEntExclude, const AABB& aabb)
+EntityVector Level::getEntities(Entity* pEntExclude, const AABB& aabb) const
 {
-	m_getEntitiesResult.clear();
+	EntityVector entities = EntityVector();
 
-	int lowerXBound = Mth::floor((aabb.min.x - 2.0f) / 16);
-	int lowerZBound = Mth::floor((aabb.min.z - 2.0f) / 16);
-	int upperXBound = Mth::floor((aabb.max.x + 2.0f) / 16);
-	int upperZBound = Mth::floor((aabb.max.z + 2.0f) / 16);
+	long lowerXBound = floor((aabb.min.x - 2.0f) / 16);
+	long lowerZBound = floor((aabb.min.z - 2.0f) / 16);
+	long upperXBound = floor((aabb.max.x + 2.0f) / 16);
+	long upperZBound = floor((aabb.max.z + 2.0f) / 16);
```

Comment:
Why did you change these to long, and why did you change it to use stl floor rather than Mth::floor?